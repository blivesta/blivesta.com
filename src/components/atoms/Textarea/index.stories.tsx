import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Textarea from '.';

const Wrapper = styled.div`
  width: 480px;
  padding: 60px;
`;

storiesOf(`atoms|Textarea`, module).add(`default`, () => (
  <Wrapper>
    <Textarea placeholder="placeholder" />
  </Wrapper>
));
