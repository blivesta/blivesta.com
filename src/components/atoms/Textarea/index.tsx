import styled from 'styled-components';
import { inputStyle } from '../Input';

const Textarea = styled.textarea`
  ${inputStyle};

  height: 200px;
  padding-top: ${props => props.theme.spaces[2]};
  padding-bottom: ${props => props.theme.spaces[2]};
  margin-bottom: ${props => props.theme.spaces[4]};
  resize: none;
`;

export default Textarea;
