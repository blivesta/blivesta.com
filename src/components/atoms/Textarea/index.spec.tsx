import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Textarea from '.';

describe('Textarea', () => {
  it('snapshot', () => {
    const tree = shallow(<Textarea />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
