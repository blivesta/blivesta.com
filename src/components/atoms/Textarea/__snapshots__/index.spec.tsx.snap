// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          "
  width: 100%;
  padding: 0 ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
  color: ",
          [Function],
          ";
  background-color: ",
          [Function],
          ";
  border-top: 0;
  border-left: 0;
  border-right: 0;
  border-bottom: 1px solid ",
          [Function],
          ";
  outline: none;
  box-shadow: none;
  transition: background, border-bottom ",
          [Function],
          ";

  &:focus {
    border-bottom: 1px solid ",
          [Function],
          ";
    background-color: ",
          [Function],
          ";
  }
",
          ";

  height: 200px;
  padding-top: ",
          [Function],
          ";
  padding-bottom: ",
          [Function],
          ";
  margin-bottom: ",
          [Function],
          ";
  resize: none;
",
        ],
      },
      "displayName": "styled.textarea",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "textarea",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
